djsonb/lookups.py

```diff
@@ -72,10 +74,27 @@ def sql(self):
                         sql_tuple = FilterTree.text_similarity_filter(rule[0], pattern, True)
                     else:
                         sql_tuple = FilterTree.text_similarity_filter(rule[0], pattern, False)
-                    rule_specs.append(sql_tuple)
+                    # add to the list of rules generated for this pattern (one per field)
+                    got_pattern = patterns.get(pattern) or []
+                    got_pattern.append(sql_tuple)
+                    patterns[pattern] = got_pattern
+
+        rule_strings = [' AND '.join([rule[0] for rule in rule_specs])]
```

Comment:
It seems confusing that rule_strings does double duty--starting out as a list of zero to two strings and then being overwritten with those same strings joined. Maybe instead of making rule_strings a list, git this first set of rules a name (exact_rules_string?), then in the `if rule_strings[0] != '' and rule_strings[1] != ''` etc sequence below just use that and `pattern_strings` and combine them into `rule_strings`.